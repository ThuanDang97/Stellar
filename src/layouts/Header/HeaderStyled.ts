import styled, { css, keyframes } from 'styled-components'

// Utils
import { pxToRem } from '@utils/theme'

// Components
import { TextStyled, ViewStyled } from '@components/styled-components'

// Props type
import { HeaderStyledProps } from '@self-types/layouts/header.props'

const fadeIn = keyframes`
from {
  opacity: 0
}

to: {
  opacity: 1
}
`

export const HeaderStyled = styled(ViewStyled)<HeaderStyledProps>`
  top: 0;

  ${({ delay }) =>
    delay &&
    css`
      animation: ${fadeIn} 2s linear both ${delay}s;
    `}

  @media (max-width: ${({ theme }) => pxToRem(theme.metrics.breakPoints.lg)}) {
    padding-left: ${(theme) => pxToRem(theme.theme.metrics.dimensions.md)};
    padding-right: ${(theme) => pxToRem(theme.theme.metrics.dimensions.md)};
  }
`

export const LogoStyled = styled(TextStyled)`
  cursor: pointer;
  @media (max-width: ${({ theme }) => pxToRem(theme.metrics.breakPoints.lg)}) {
    margin-right: ${(theme) => pxToRem(theme.theme.metrics.dimensions.md)};
  }
`

export const WrapperAvatarStyled = styled(ViewStyled)`
  @media (max-width: ${({ theme }) => pxToRem(theme.metrics.breakPoints.lg)}) {
    column-gap: ${(theme) => pxToRem(theme.theme.metrics.dimensions.xs)};
  }
`
