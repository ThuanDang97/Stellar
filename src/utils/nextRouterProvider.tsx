import { NextRouter } from 'next/router'
import { RouterContext } from 'next/dist/shared/lib/router-context'

const createMockRouter = (router?: Partial<NextRouter>): NextRouter => ({
  basePath: '',
  pathname: '/',
  route: '/',
  query: {},
  asPath: '/',
  back: jest.fn(),
  beforePopState: jest.fn(),
  prefetch: jest.fn(),
  push: jest.fn(),
  reload: jest.fn(),
  replace: jest.fn(),
  events: {
    on: jest.fn(),
    off: jest.fn(),
    emit: jest.fn(),
  },
  isFallback: false,
  isLocaleDomain: false,
  isReady: true,
  defaultLocale: 'en',
  domainLocales: [],
  isPreview: false,
  ...router,
})

export const NextRouterProvider = ({
  children,
  router = { query: {} },
}: {
  children: React.ReactNode
  router: Partial<NextRouter>
}) => (
  <RouterContext.Provider value={createMockRouter(router)}>
    {children}
  </RouterContext.Provider>
)
