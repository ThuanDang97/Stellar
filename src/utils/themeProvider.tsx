import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
