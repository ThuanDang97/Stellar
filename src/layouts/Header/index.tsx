import { memo, useCallback, useEffect, useMemo, useRef, useState } from 'react'
import Image from 'next/image'
import { useTheme } from 'styled-components'
import { useRouter } from 'next/router'

// Constants
import {
  HEADER_NAVIGATION,
  HEADER_TYPES,
  LOCAL_STORAGE_KEY,
  MENU_STATUS,
  PRIMARY_HEADER_URL,
  SECONDARY_URL,
} from '@constants/index'

// Props type
import { HeaderProps } from '@self-types/index'
import { NotificationItemProps } from '@self-types/components/Notification.props'

// Mocks
import { NOTIFICATIONS } from '@mocks/mockData'

// Utils
import { getLocalStorage } from '@utils/localStorage'

// Components
import HeaderNav from '@components/HeaderNav'
import Avatar from '@components/Avatar'
import Button from '@components/Button'
import LinkComponent from '@components/Link'
import { Container, ViewStyled } from '@components/styled-components'
import MenuProfile from '@components/MenuProfile'
import Notifications from '@components/Notifications'
import { HeaderStyled, LogoStyled, WrapperAvatarStyled } from './HeaderStyled'

const Header = ({
  HeaderType,
  avatarUrl,
  userFullName = '',
  bgColor,
  showArrow,
  loginHref,
  pLeft,
  pRight,
  pTop,
  pBottom,
  isLandingPage,
  position,
  isArrowLight,
  zIndex,
}: HeaderProps) => {
  const router = useRouter()
  const theme = useTheme()
  const wrapperRef = useRef<HTMLInputElement>(null)
  const wrapperRefNotif = useRef<HTMLInputElement>(null)
  const [menuStatus, setMenuStatus] = useState<string>('none')
  const [notifPopupStatus, setNotifPopupStatus] = useState<string>('none')
  const [count, setCount] = useState(NOTIFICATIONS.length)
  const [idNotif, setIdNotification] = useState('')
  const [innerWidth, setInnerWidth] = useState(window.screen.width)

  const handleGoBack = () => {
    const hasAccount = getLocalStorage(LOCAL_STORAGE_KEY.IS_TOKEN)
    if (!hasAccount) {
      router.push(SECONDARY_URL.LANDING_PAGE.URL)
    } else {
      router.back()
    }
  }

  // Go to dashboard page when click logo on header
  const handleGoHomePage = () => {
    router.push(PRIMARY_HEADER_URL.DASHBOARD.URL)
    if (router.pathname === PRIMARY_HEADER_URL.DASHBOARD.URL) router.reload()
  }

  useEffect(() => {
    /**
     * Alert if clicked on outside of element
     */
    const handleClickOutside = (event: MouseEvent) => {
      if (
        wrapperRef &&
        wrapperRef.current &&
        !wrapperRef.current.contains(event.target as Node)
      ) {
        if (menuStatus === MENU_STATUS.FADE_IN) {
          setMenuStatus(MENU_STATUS.NONE)
        }
      }
    }
    // Bind the event listener
    document.addEventListener('mousedown', handleClickOutside)
    return () => {
      // Unbind the event listener on clean up
      document.removeEventListener('mousedown', handleClickOutside)
    }
  }, [wrapperRef, menuStatus])

  useEffect(() => {
    /**
     * Alert if clicked on outside of element
     */
    const handleClickOutside = (event: MouseEvent) => {
      if (
        wrapperRefNotif &&
        wrapperRefNotif.current &&
        !wrapperRefNotif.current.contains(event.target as Node)
      ) {
        if (notifPopupStatus === MENU_STATUS.FADE_IN) {
          setNotifPopupStatus(MENU_STATUS.NONE)
        }
      }
    }
    // Bind the event listener
    document.addEventListener('mousedown', handleClickOutside)
    return () => {
      // Unbind the event listener on clean up
      document.removeEventListener('mousedown', handleClickOutside)
    }
  }, [wrapperRefNotif, notifPopupStatus])

  useEffect(() => {
    const handleResize = () => {
      setInnerWidth(window.screen.width)
    }
    // Add event  and remove event resize screen
    window.addEventListener('resize', handleResize)
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  // Show menu when click avatar on header
  const handleToggleMenu = () => {
    if (menuStatus === MENU_STATUS.FADE_IN) {
      setMenuStatus(MENU_STATUS.FADE_OUT)
    } else {
      setMenuStatus(MENU_STATUS.FADE_IN)
    }
  }

  // Show notification when click icon notification
  const handleToggleNotifications = () => {
    setCount(0)
    if (notifPopupStatus === MENU_STATUS.FADE_IN) {
      setNotifPopupStatus(MENU_STATUS.FADE_OUT)
    } else {
      setNotifPopupStatus(MENU_STATUS.FADE_IN)
    }
  }

  // Handle Get Id Notification
  const handleUpdateNotificationId = useCallback(
    (id: string) => {
      setIdNotification(id)
    },
    [idNotif],
  )

  const listNotificationUpdate: NotificationItemProps[] = useMemo(() => {
    NOTIFICATIONS.forEach((item) => {
      if (item.id === idNotif) {
        NOTIFICATIONS[Number(idNotif) - 1].isCheck = true
      }
    })
    return NOTIFICATIONS
  }, [idNotif])

  const isMobile = innerWidth <= theme.metrics.breakPoints.xs

  return (
    <HeaderStyled
      as="header"
      position={position}
      width={theme.metrics.width.full}
      pLeft={pLeft ?? theme.metrics.dimensions.xxl}
      pRight={pRight ?? theme.metrics.dimensions.xxl}
      pTop={pTop ?? theme.metrics.dimensions.sm}
      pBottom={pBottom ?? theme.metrics.dimensions.sm}
      bgColor={bgColor}
      zIndex={zIndex}
      {...(isLandingPage && { delay: 1 })}
    >
      <Container
        textAlign="center"
        display="flex"
        justifyContent="space-between"
        maxWidth={theme.metrics.breakPoints.xxl}
        margin="0 auto"
      >
        {HeaderType === HEADER_TYPES.DEFAULT && (
          <>
            <ViewStyled display="flex" justifyContent="flex-start" width="15%">
              {showArrow && (
                <Button
                  variant="icon"
                  height={theme.metrics.dimensions.md}
                  width={theme.metrics.dimensions.md}
                  imgUrl={
                    isArrowLight
                      ? '/icons/chevron-left-light.svg'
                      : '/icons/chevron-left.svg'
                  }
                  onClick={handleGoBack}
                />
              )}
            </ViewStyled>
            <ViewStyled width="70%">
              <Image
                height={theme.metrics.dimensions.lg}
                width={theme.metrics.dimensions.lg}
                src="/images/logo.svg"
              />
            </ViewStyled>
            <ViewStyled display="flex" justifyContent="flex-end" width="15%">
              {loginHref && (
                <LinkComponent
                  width={theme.metrics.width.common}
                  linkTypes="loginLink"
                  href={loginHref}
                  text="Login"
                  {...(isLandingPage && { color: theme.colors.white })}
                />
              )}
            </ViewStyled>
          </>
        )}
        {HeaderType === HEADER_TYPES.PRIMARY && (
          <>
            <ViewStyled display="flex">
              <LogoStyled
                display="flex"
                alignItems="center"
                mRight={theme.metrics.dimensions.xxl}
                onClick={handleGoHomePage}
              >
                <Image
                  src="/images/stellar.png"
                  width={theme.metrics.width.xmd}
                  height={theme.metrics.height.default}
                />
              </LogoStyled>
              {!isMobile && <HeaderNav headerNavList={HEADER_NAVIGATION} />}
            </ViewStyled>
            <WrapperAvatarStyled
              columnGap={theme.metrics.dimensions.base}
              display="flex"
              position="relative"
            >
              <ViewStyled ref={wrapperRefNotif}>
                <Button
                  variant="light"
                  onClick={handleToggleNotifications}
                  imgUrl="/icons/bell.svg"
                  width={theme.metrics.icons.default}
                  height={theme.metrics.icons.default}
                  size="medium"
                  count={count}
                  mTop={theme.metrics.dimensions.sm}
                />

                <Notifications
                  notifPopupStatus={notifPopupStatus}
                  notificationList={listNotificationUpdate}
                  handleUpdateNotificationId={handleUpdateNotificationId}
                />
              </ViewStyled>
              <ViewStyled ref={wrapperRef}>
                <Avatar
                  imageUrl={avatarUrl}
                  {...(!avatarUrl &&
                    userFullName && { fullName: userFullName })}
                  handleToggleMenu={handleToggleMenu}
                />
                <MenuProfile
                  menuStatus={menuStatus}
                  setMenuStatus={setMenuStatus}
                  fullName={userFullName}
                  imageUrl={avatarUrl}
                />
              </ViewStyled>
            </WrapperAvatarStyled>
          </>
        )}
      </Container>
      {isMobile && HeaderType === HEADER_TYPES.PRIMARY && (
        <HeaderNav isMobile={isMobile} headerNavList={HEADER_NAVIGATION} />
      )}
    </HeaderStyled>
  )
}

export default memo(Header)
