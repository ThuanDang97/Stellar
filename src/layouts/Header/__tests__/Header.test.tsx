// Layouts
import Header from '@layouts/Header'

// Themes
import { theme } from '@themes/index'

// Utils
import { renderWithTheme } from '@utils/themeProvider'
import { NextRouterProvider } from '@utils/nextRouterProvider'

const route = { route: '/abc' }

describe('Header layout', () => {
  const defaultComponent = (
    <NextRouterProvider router={route}>
      <Header showArrow loginHref="/login" HeaderType="defaultHeader" />
    </NextRouterProvider>
  )

  const primaryHeader = (
    <NextRouterProvider router={route}>
      <Header
        HeaderType="primaryHeader"
        avatarUrl="https://picsum.photos/20"
        bgColor={theme.colors.firefly}
      />
    </NextRouterProvider>
  )

  test('Header Default matchers DOM Snapshot', () => {
    const { container } = renderWithTheme(defaultComponent)

    expect(container).toMatchSnapshot()
  })

  test('Header Primary matchers DOM Snapshot', () => {
    const { container } = renderWithTheme(primaryHeader)

    expect(container).toMatchSnapshot()
  })
})
