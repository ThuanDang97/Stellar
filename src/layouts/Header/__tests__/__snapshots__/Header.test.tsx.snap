// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header layout Header Default matchers DOM Snapshot 1`] = `
<div>
  <header
    class="ViewStyled-sc-rqe5sa-0 HeaderStyled-sc-ht65yv-0 fGSqOK fKKWJQ"
    width="100%"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__Container-sc-rqe5sa-9 ivbjTx dTVWqF"
      display="flex"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 eqYeCy"
        display="flex"
        width="15%"
      >
        <button
          class="ButtonStyled-sc-1ys9v52-0 dFjjMT"
          data-testid="button"
        >
          <span
            class="ButtonStyled__ButtonIconStyled-sc-1ys9v52-1 DjHyU"
          >
            <div
              class="ViewStyled-sc-rqe5sa-0 lFLaW"
              display="inline-block"
              height="20"
              width="20"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
          </span>
        </button>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 hItHKT"
        width="70%"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2730%27%20height=%2730%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 fMTQRR"
        display="flex"
        width="15%"
      >
        <a
          class="LinkStyled__ButtonLinkStyled-sc-626068-1 gswOa"
          data-testid="link"
          href="/login"
          width="50"
        >
          Login
        </a>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header layout Header Primary matchers DOM Snapshot 1`] = `
<div>
  <header
    class="ViewStyled-sc-rqe5sa-0 HeaderStyled-sc-ht65yv-0 eEKeVh fKKWJQ"
    width="100%"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__Container-sc-rqe5sa-9 ivbjTx dTVWqF"
      display="flex"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 knVWgS"
        display="flex"
      >
        <span
          class="TextStyled-sc-9xia41-0 HeaderStyled__LogoStyled-sc-ht65yv-1 bUlkxO hvoHPo"
          display="flex"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27105%27%20height=%2720%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </span>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 HeaderStyled__WrapperAvatarStyled-sc-ht65yv-2 iRjzkB faxsfr"
        display="flex"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 fTsvHV"
        >
          <button
            class="ButtonStyled-sc-1ys9v52-0 dDdKTh"
            data-testid="button"
          >
            <span
              class="ButtonStyled__ButtonIconStyled-sc-1ys9v52-1 DjHyU"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 lFLaW"
                display="inline-block"
                height="20"
                width="20"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
                <span
                  class="TextStyled-sc-9xia41-0 TextStyled__BadgeNotifStyled-sc-9xia41-5 gtHdGO hQwyqw"
                >
                  4
                </span>
              </div>
            </span>
          </button>
          <div
            class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 ViewStyled__BoxShadowAnimationStyled-sc-rqe5sa-3 hgketM csavJt hhfTyz"
            data-testid="notificationsHeader"
            width="300"
          >
            <ul
              class="ListStyled-sc-t03hvr-0 gYEiwT"
              display="none"
              width="100%"
            >
              <li
                class="ListStyled__ListItemStyled-sc-t03hvr-1 kGAnkc"
              >
                <div
                  class="ViewStyled-sc-rqe5sa-0 XoSwA"
                  display="flex"
                >
                  <span
                    class="TextStyled-sc-9xia41-0 jfztUt"
                    font-size="16"
                    font-weight="700"
                  >
                    Credit Score Update
                  </span>
                  <span
                    class="TextStyled-sc-9xia41-0 TextStyled__TextDescriptionStyled-sc-9xia41-2 iixUHT kyJdLf"
                    color="#242A30"
                    overflow="hidden"
                  >
                    Your credit score increased by...
                  </span>
                  <span
                    class="TextStyled-sc-9xia41-0 TextStyled__TextDescriptionStyled-sc-9xia41-2 cntApC kyJdLf"
                    color="#242A30"
                  >
                    1 hours
                  </span>
                </div>
              </li>
              <li
                class="ListStyled__ListItemStyled-sc-t03hvr-1 kGAnkc"
              >
                <div
                  class="ViewStyled-sc-rqe5sa-0 XoSwA"
                  display="flex"
                >
                  <span
                    class="TextStyled-sc-9xia41-0 jfztUt"
                    font-size="16"
                    font-weight="700"
                  >
                    Bill Pay
                  </span>
                  <span
                    class="TextStyled-sc-9xia41-0 TextStyled__TextDescriptionStyled-sc-9xia41-2 iixUHT kyJdLf"
                    color="#242A30"
                    overflow="hidden"
                  >
                    We noticed you missed a payment...
                  </span>
                  <span
                    class="TextStyled-sc-9xia41-0 TextStyled__TextDescriptionStyled-sc-9xia41-2 cntApC kyJdLf"
                    color="#242A30"
                  >
                    3 days ago
                  </span>
                </div>
              </li>
              <li
                class="ListStyled__ListItemStyled-sc-t03hvr-1 kGAnkc"
              >
                <div
                  class="ViewStyled-sc-rqe5sa-0 XoSwA"
                  display="flex"
                >
                  <span
                    class="TextStyled-sc-9xia41-0 jfztUt"
                    font-size="16"
                    font-weight="700"
                  >
                    Offer
                  </span>
                  <span
                    class="TextStyled-sc-9xia41-0 TextStyled__TextDescriptionStyled-sc-9xia41-2 iixUHT kyJdLf"
                    color="#242A30"
                    overflow="hidden"
                  >
                    Look like you may quality for...
                  </span>
                  <span
                    class="TextStyled-sc-9xia41-0 TextStyled__TextDescriptionStyled-sc-9xia41-2 cntApC kyJdLf"
                    color="#242A30"
                  >
                    10 days ago
                  </span>
                </div>
              </li>
              <li
                class="ListStyled__ListItemStyled-sc-t03hvr-1 kGAnkc"
              >
                <div
                  class="ViewStyled-sc-rqe5sa-0 XoSwA"
                  display="flex"
                >
                  <span
                    class="TextStyled-sc-9xia41-0 jfztUt"
                    font-size="16"
                    font-weight="700"
                  >
                    Offer
                  </span>
                  <span
                    class="TextStyled-sc-9xia41-0 TextStyled__TextDescriptionStyled-sc-9xia41-2 iixUHT kyJdLf"
                    color="#242A30"
                    overflow="hidden"
                  >
                    Look like you may quality for offer
                  </span>
                  <span
                    class="TextStyled-sc-9xia41-0 TextStyled__TextDescriptionStyled-sc-9xia41-2 cntApC kyJdLf"
                    color="#242A30"
                  >
                    11 days ago
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="ViewStyled-sc-rqe5sa-0 fTsvHV"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 iYPAli"
            cursor="pointer"
            height="40"
            width="40"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                class="AvatarStyled__AvatarImageStyled-sc-1ppokaq-0 WGkXw"
                data-nimg="fill"
                data-testid="imgAvatar"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 ViewStyled__BoxShadowAnimationStyled-sc-rqe5sa-3 exDRui csavJt hhfTyz"
            data-testid="menuHeader"
            width="300"
          >
            <ul
              class="ListStyled-sc-t03hvr-0 hFGRnl"
              width="100%"
            >
              <li
                class="ListStyled__ListItemStyled-sc-t03hvr-1 ggeHjz"
              >
                <div
                  class="ViewStyled-sc-rqe5sa-0 HeaderStyled__WrapperAvatarStyled-sc-ht65yv-2 fkNXGy faxsfr"
                  display="flex"
                >
                  <div
                    class="ViewStyled-sc-rqe5sa-0 dEUOMm"
                    cursor="pointer"
                    height="60"
                    width="60"
                  >
                    <span
                      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                    >
                      <img
                        class="AvatarStyled__AvatarImageStyled-sc-1ppokaq-0 WGkXw"
                        data-nimg="fill"
                        data-testid="imgAvatar"
                        decoding="async"
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                      />
                      <noscript />
                    </span>
                  </div>
                  <div
                    class="ViewStyled-sc-rqe5sa-0 iMoZMg"
                    display="flex"
                  >
                    <span
                      class="TextStyled-sc-9xia41-0 evMDQc"
                      font-weight="700"
                      overflow="hidden"
                    />
                    <a
                      class="LinkStyled__ButtonLinkStyled-sc-626068-1 fbACSi"
                      color="#101F30"
                      data-testid="link"
                      display="inline"
                      font-family="AdobeCleanLight"
                      font-size="13"
                      href="/profile"
                      letter-spacing="0.9"
                    >
                      See Your Profile
                    </a>
                  </div>
                </div>
              </li>
              <li
                class="ListStyled__ListItemStyled-sc-t03hvr-1 dBsbXq"
                href="/refer-a-friend"
              >
                <div
                  class="ViewStyled-sc-rqe5sa-0 kQGDDR"
                >
                  <a
                    class="LinkStyled__ButtonLinkStyled-sc-626068-1 iIGnIj"
                    data-testid="link"
                    display="inline"
                    font-size="16"
                    font-weight="700"
                    href="/refer-a-friend"
                  >
                    Refer A Friend
                  </a>
                </div>
              </li>
              <li
                class="ListStyled__ListItemStyled-sc-t03hvr-1 dBsbXq"
                href="/connect-bank?returnUrl=%2Fabc"
              >
                <div
                  class="ViewStyled-sc-rqe5sa-0 kQGDDR"
                >
                  <a
                    class="LinkStyled__ButtonLinkStyled-sc-626068-1 iIGnIj"
                    data-testid="link"
                    display="inline"
                    font-size="16"
                    font-weight="700"
                    href="/connect-bank?returnUrl=%2Fabc"
                  >
                    Connect Bank
                  </a>
                </div>
              </li>
              <li
                class="ListStyled__ListItemStyled-sc-t03hvr-1 eXKxno"
              >
                <ul
                  class="ListStyled-sc-t03hvr-0 jAJdQE"
                  display="block"
                  width="100%"
                >
                  <li
                    class="ListStyled__ListItemStyled-sc-t03hvr-1 koFPQW"
                    href="/dashboard"
                  >
                    <a
                      class="LinkStyled__ButtonLinkStyled-sc-626068-1 fyaZWH"
                      data-testid="link"
                      display="inline"
                      font-size="16"
                      font-weight="700"
                      href="/dashboard"
                    >
                      Dashboard
                    </a>
                  </li>
                  <li
                    class="ListStyled__ListItemStyled-sc-t03hvr-1 koFPQW"
                  >
                    <a
                      class="LinkStyled__ButtonLinkStyled-sc-626068-1 fyaZWH"
                      data-testid="link"
                      display="inline"
                      font-size="16"
                      font-weight="700"
                      href="/help"
                    >
                      Help & Support
                    </a>
                  </li>
                  <li
                    class="ListStyled__ListItemStyled-sc-t03hvr-1 koFPQW"
                  >
                    <button
                      class="ButtonStyled-sc-1ys9v52-0 gwnvOP"
                      color="#242A30"
                      data-testid="buttonLogout"
                      font-size="16"
                      font-weight="700"
                    >
                      Log Out
                    </button>
                  </li>
                </ul>
              </li>
              <li
                class="ListStyled__ListItemStyled-sc-t03hvr-1 jfbuwf"
              >
                <span
                  class="TextStyled-sc-9xia41-0 jjPAoQ"
                >
                  <a
                    class="LinkStyled__ButtonLinkStyled-sc-626068-1 bvoeHr"
                    color="#919099"
                    data-testid="link"
                    display="inline"
                    font-family="AdobeCleanLight"
                    font-size="13"
                    href="/privacy-policy"
                    letter-spacing="0.4"
                  >
                    Privacy Policy
                  </a>
                  <span
                    class="TextStyled-sc-9xia41-0 hrVQLR"
                    color="#919099"
                  >
                     - 
                  </span>
                  <a
                    class="LinkStyled__ButtonLinkStyled-sc-626068-1 bvoeHr"
                    color="#919099"
                    data-testid="link"
                    display="inline"
                    font-family="AdobeCleanLight"
                    font-size="13"
                    href="/terms-of-use"
                    letter-spacing="0.4"
                  >
                    Terms of Use
                  </a>
                </span>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <ul
      class="ListStyled-sc-t03hvr-0 cYiWEp"
    >
      <li
        class="ListStyled__ListItemStyled-sc-t03hvr-1 GyEBw"
        data-testid="item1"
      >
        <a
          class="LinkStyled-sc-626068-0 fqiYnA"
          color="#FFF"
          data-testid="/help-1"
          href="/help"
        >
          Help
        </a>
      </li>
      <li
        class="ListStyled__ListItemStyled-sc-t03hvr-1 GyEBw"
        data-testid="item2"
      >
        <a
          class="LinkStyled-sc-626068-0 fqiYnA"
          color="#FFF"
          data-testid="/feedback-2"
          href="/feedback"
        >
          Feedback
        </a>
      </li>
      <li
        class="ListStyled__ListItemStyled-sc-t03hvr-1 GyEBw"
        data-testid="item3"
      >
        <a
          class="LinkStyled-sc-626068-0 fqiYnA"
          color="#FFF"
          data-testid="/refer-a-friend-3"
          href="/refer-a-friend"
        >
          Refer A Friend
        </a>
      </li>
    </ul>
  </header>
</div>
`;
